mkosi/architecture.py

```diff
@@ -0,0 +1,133 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import enum
+import platform
+
+from mkosi.log import die
+
+
+class Architecture(enum.Enum):
+    alpha       = "alpha"
+    arc         = "arc"
+    arm         = "arm"
+    arm64       = "arm64"
+    ia64        = "ia64"
+    loongarch64 = "loongarch64"
+    mips_le     = "mips-le"
+    mips64_le   = "mips64-le"
+    parisc      = "parisc"
+    ppc         = "ppc"
+    ppc64       = "ppc64"
+    ppc64_le    = "ppc64-le"
+    riscv32     = "riscv32"
+    riscv64     = "riscv64"
+    s390        = "s390"
+    s390x       = "s390x"
+    tilegx      = "tilegx"
+    x86         = "x86"
+    x86_64      = "x86-64"
+
+    def __str__(self) -> str:
+        return self.value
+
+    @staticmethod
+    def from_uname(s: str) -> "Architecture":
```

Comment:
```suggestion
    @classmethod
    def from_uname(cls, s: str) -> "Architecture":
        if not (a := getattr(cls, s)):
            die(f"Architecture {a} is not supported")
        return a
```
would be a possibility if the unames were entered as aliases above, e.g.
```python

class Architecture(enum.Enum):
    alpha       = "alpha"
    arc         = "arc"
    arm         = "arm"
    arm64       = "arm64"
    ia64        = "ia64"
    loongarch64 = "loongarch64"
    mips_le     = "mips-le"
    mips64_le   = "mips64-le"
    parisc      = "parisc"
    ppc         = "ppc"
    ppc64       = "ppc64"
    ppc64_le    = "ppc64-le"
    riscv32     = "riscv32"
    riscv64     = "riscv64"
    s390        = "s390"
    s390x       = "s390x"
    tilegx      = "tilegx"
    x86         = "x86"
    x86_64      = "x86-64"

    # aliases
    aarch64 = "arm64"
    aarch64_be = "arm64"
```

They are not iterated over and also don't appear in most tab completions, but `Architecture.aarch64` would resolve to `Architecture.arm64` right away. Unfortunately it doesn't come with a string constructor out of the box, but it would remove a bit of the repetition.